apps/journeys-admin/src/components/JourneyList/JourneyCard/JourneyCardMenu/DefaultMenu/ArchiveJourney/ArchiveJourney.tsx

```diff
@@ -82,21 +86,36 @@ export function ArchiveJourney({
 
   const { enqueueSnackbar } = useSnackbar()
 
-  async function handleClick(): Promise<void> {
+  async function handleArchive(): Promise<void> {
+    setLoading(true)
     try {
-      if (status !== JourneyStatus.archived) {
-        await archiveJourney()
-        enqueueSnackbar(t('Journey Archived'), {
-          variant: 'success',
-          preventDuplicate: true
-        })
-      } else {
-        await unarchiveJourney()
-        enqueueSnackbar(t('Journey Unarchived'), {
-          variant: 'success',
+      await archiveJourney()
+      enqueueSnackbar(t('Journey Archived'), {
+        variant: 'success',
+        preventDuplicate: true
+      })
+      await refetch?.()
+      handleClose()
+    } catch (error) {
+      if (error instanceof Error) {
+        enqueueSnackbar(error.message, {
+          variant: 'error',
           preventDuplicate: true
         })
       }
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  async function handleUnarchive(): Promise<void> {
+    setLoading(true)
+    try {
```

Comment:
remove try catch and use apollo's in built error handling.
Do this for both the handleArchive and handleUnarchive

```suggestion
         await unarchiveJourney({
        onError: () => {
          enqueueSnackbar(t('Journey Unarchived'), {
            variant: 'success',
            preventDuplicate: true
          })
          setLoading(false)
        },
        onCompleted: async () => {
          enqueueSnackbar(t('Journey Unarchived'), {
            variant: 'success',
            preventDuplicate: true
          })
          await refetch?.()
          handleClose()
          setLoading(false)
        }
      })
```

Reply:
Done